vars/capsulePipelineWrapper.groovy

```diff
@@ -1,245 +1,485 @@
+
+        vegacapsule template node-sets \
+            --home-path '""" + netHome + """' \
+            --path '""" + tmplHome + """/tendermint.common.tmpl.toml' \
+            --nodeset-group-name full \
+            --type tendermint \
+            --with-merge """ + additionalFlags + """
+
+        vegacapsule template node-sets \
+            --home-path '""" + netHome + """' \
+            --path '""" + tmplHome + """/data_node.full.tmpl.toml' \
+            --nodeset-group-name full \
+            --type data-node \
+            --with-merge """ + additionalFlags
+}
+
+void getReleasedBinary(String ghCredentialsID, String binType, String version) {
+  withGHCLI('credentialsId': ghCredentialsID) {
+        sh 'gh release --repo vegaprotocol/vega download ' + version + ' --pattern "' + binType + '-linux*"'
+  }
+
+  sh 'mv vega-linux-amd64 bin/' + binType
+  sh 'chmod +x bin/' + binType
+  sh binType + ' version'
+}
+
+void compileCoreBinary(String ghCredentialsID, String binType, String version, String outDir) {
+  Map packagesMapping = [
+        'vega': 'cmd/vega/',
+        'vegawallet': 'cmd/vegawallet/',
+        'data-node': 'cmd/data-node/'
+    ]
+
+  if (!packagesMapping.containsKey(binType)) {
+    error('[compileBinary] invalid binary type. got: "' + binType + '"')
+  }
+
+  gitClone([
+      url: 'git@github.com:vegaprotocol/vegacapsule.git',
+      branch: 'main',
+      directory: version,
+      credentialsId: ghCredentialsID
+  ])
+  sh label: 'Building ' + binType,
+        script: 'go mod vendor && go build -o ' + outDir + '/' + binType + ' ' + packagesMapping[binType]
+}
+
+void downloadS3Binary(String s3Path, String binName, String outDir) {
+  sh label: 'Download the ' + binName + ' from the S3',
+       script: 'aws s3 cp --only-show-errors --no-progress "' + s3Path + '" "' + outDir + '/' + binName + '"'
+  sh label: 'Set permissions on the ' + binName + ' binary',
+       script: 'chmod a+x "' + outDir + '/' + binName + '"'
+}
+
+boolean isS3Link(String path) {
+    /* groovylint-disable-next-line DuplicateNumberLiteral */
+  return path.length() > 0 && path[0..4] == 's3://'
+}
+
+void veganetSh(String network, String command, String sshCredentialsId, String dockerCredentialsId) {
+  dir('devops-infra') {
+    withCredentials([sshUserPrivateKey(
+        credentialsId: sshCredentialsId,
+        keyFileVariable: 'PSSH_KEYFILE',
+        usernameVariable: 'PSSH_USER'
+    )]) {
+      withDockerRegistry([
+          credentialsId: dockerCredentialsId,
+          url: 'https://ghcr.io'
+      ]) {
+        withGoogleSA('gcp-k8s') {
+          sh label: 'Create markets',
+            script: './veganet.sh ' + network + ' ' + command
+        }
+      }
     }
+  }
+}
+
+void call(Map customConfig = [:]) {
+  Map config = [
+    networkName: '',
+    nomadAddress: '',
+    awsRegion: '',
+    vegacapsuleS3BucketName: '',
+  ] + customConfig
+
+  pipeline {
+    agent any
```

Comment:
Maybe it's worth asking ourselves question if there's any specific agent label we want to use for capsule

Reply:
no, there is no agent. We have everything on standard linux server :)